tests/statecheck/util.go

```diff
@@ -42,23 +42,44 @@ func newSdkConnection(ctx context.Context) (connection.Connection, error) {
 func snapshotBackends(target *postgres.Client, analyzer string, tables []string) (int64, error) {
 	ctx := context.Background()
 
+	schema := fmt.Sprintf("snapshot_%s", analyzer)
+
 	batch := &storage.QueryBatch{}
-	batch.Queue(`CREATE SCHEMA IF NOT EXISTS snapshot;`)
+	batch.Queue(fmt.Sprintf(`CREATE SCHEMA IF NOT EXISTS %s;`, schema))
 	for _, t := range tables {
 		batch.Queue(fmt.Sprintf(`
-			DROP TABLE IF EXISTS snapshot.%s CASCADE;
-		`, t))
+			DROP TABLE IF EXISTS %s.%s CASCADE;
+		`, schema, t))
 		batch.Queue(fmt.Sprintf(`
-			CREATE TABLE snapshot.%[1]s AS TABLE chain.%[1]s;
-		`, t))
+			CREATE TABLE %s.%[2]s AS TABLE chain.%[2]s;
+		`, schema, t))
 	}
-	batch.Queue(`
-		INSERT INTO snapshot.snapshotted_heights (analyzer, height)
+
+	// Store the height at which the snapshot was taken.
+	batch.Queue(fmt.Sprintf(`
+		CREATE TABLE IF NOT EXISTS %s.snapshotted_heights (
```

Comment:
Agree, it was especially weird seeing it in the migrations. Will remove it.

Reply:
> (I know we query it at the end of this func, but we could just directly query analysis_processed_blocks instead)

I think we need at least a temporary table to store the height since we don't return query responses in `SendBatchWithOptions`, but the query for the height should run in the same transaction as the rest of the queries. Otherwise the reported height might be wrong (increased already since the snapshot was done). 

I'll simplify to creating a temporary table to store the height for now.